src/main/java/uk/gov/justice/hmpps/prison/aop/connectionproxy/ResettableContextConnection.kt

```diff
@@ -0,0 +1,22 @@
+package uk.gov.justice.hmpps.prison.aop.connectionproxy
+
+import org.slf4j.LoggerFactory
+import java.sql.Connection
+import java.sql.SQLException
+
+class ResettableContextConnection(private val connection: Connection) : Connection by connection {
+
+  @Throws(SQLException::class)
+  override fun close() {
+    log.debug("Closing context clearing connection {}", this)
+    resetContext()
+    connection.close()
+  }
+
+  @Throws(SQLException::class)
+  fun resetContext(): Boolean = connection.run("BEGIN nomis_context.set_nomis_context(); END;")
```

Comment:
This has been tested against T3 and resets the connection's NOMIS context as expected.